packages/odata-service-writer/src/index.ts

```diff
@@ -71,101 +155,146 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     const paths = await findProjectFiles(basePath, fs);
```

Comment:
_ESLint: Refactor this function to reduce its Cognitive Complexity from 24 to the 15 allowed.(sonarjs/ cognitive-complexity)_